File: app/schemas/experiment.py
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

class ExperimentResponse(BaseModel):
    id: uuid.UUID
    hypothesis_id: uuid.UUID
    status: str
    experiment_dir: str
    container_id: Optional[str]
    results: Optional[dict]
    result_summary: Optional[str]
    wandb_run_url: Optional[str]
    mlflow_run_id: Optional[str]
    error_log: Optional[str]
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    created_at: datetime
    
class ExperimentListResponse(BaseModel):
    items: list[ExperimentResponse]
    total: int

File: scientific-copilot-ui/lib/types/experiment.ts
export type ExperimentStatus = 'queued' | 'running' | 'completed' | 'failed'

export interface Experiment {
  id: string
  hypothesis_id: string
  status: ExperimentStatus
  experiment_dir: string
  container_id: string | null
  results: Record<string, number | string> | null
  result_summary: string | null
  wandb_run_url: string | null
  mlflow_run_id: string | null
  error_log: string | null
  started_at: string | null
  completed_at: string | null
  created_at: string
}

export interface ExperimentResult {
  id: string
  experiment_id: string
  hypothesis_id: string
  outcome: 'validated' | 'failed' | 'inconclusive'
  metrics: Record<string, number>
  result_summary: string
  lessons_learned: string[]
  created_at: string
}

export interface PaginatedExperiments {
  items: Experiment[]
  total: number
}
